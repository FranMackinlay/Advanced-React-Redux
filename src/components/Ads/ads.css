.content-container {
	width: 100vw;
}

a {
	border-radius: inherit;
}
ul {
	text-align: center;
	list-style-type: none;
	margin: 0;
	padding: 0;
	display: grid;
	gap: 1rem;
	/* grid-template-columns: auto auto auto; */
	grid-template-columns: repeat(auto-fit, minmax(300px, 1fr));
}

.ads-container {
	overflow: scroll;
	height: 78vh;
}

@media (max-width: 400px) {
	.content-container {
		width: fit-content;
	}
	.ads-form {
		width: 290px;
		padding: 0;
		justify-content: space-around;
	}
	.select-form,
	.input-form,
	.search-form,
	.reset-form {
		margin: 3px;
	}
	.select-form {
		width: 70px;
	}
	.input-form {
		width: 80px;
	}
}

@media (max-width: 715px) {
	.create-ad {
		top: 10px;
		right: -20px;
	}
}
