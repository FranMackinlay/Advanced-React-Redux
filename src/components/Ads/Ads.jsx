import React, { useEffect } from 'react';
import Header from '../Header/header';
import Card from '../Card/card';
import './ads.css';
import { fetchAds } from '../../store/actions';
import { useDispatch } from 'react-redux';

export default function Ads(props) {
	const renderAdList = adsList => adsList.map(ad => <Card key={ad._id} ad={ad} />);

	const dispatch = useDispatch();

	useEffect(() => {
		dispatch(fetchAds());
	}, [dispatch]);

	const isLogged = localStorage.getItem('isLogged');
	if (!isLogged) props.history.push('/login');
	if (props.ads === []) return <h1>Loading Ads...</h1>;
	return (
		<div className='content-container'>
			<Header />
			<div className='ads-container'>
				<ul>{renderAdList(props.ads)}</ul>
			</div>
		</div>
	);
}
