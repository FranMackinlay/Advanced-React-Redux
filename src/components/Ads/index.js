import { connect } from 'react-redux';

import Ads from './Ads';

import { getVisibleAds, localGetter } from '../../store/selectors';

function mapStateToProps(state, ownProps) {
  console.log('STATE', state);
  return {
    ads: getVisibleAds(state),
    localCheck: localGetter(state),
  };
}

export default connect(mapStateToProps)(Ads);
