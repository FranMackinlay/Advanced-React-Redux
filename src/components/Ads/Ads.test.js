import React from 'react';
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import Ads from './Ads';

const history = {
  push: jest.fn(),
};

describe('Ads', () => {
  const store = {
    dispatch: () => { },
    getState: () => ({
      ads: [
        {
          name: 'No Man\'s Sky',
          price: '300',
          description: 'Best game ever',
          createdAt: '1/1/11',
          tags: 'lifestyle,mobile',
          type: 'buy',
          updatedAd: '2/1/11',
          _id: '56gc3h45v4bj4h5',
        }
      ],
    }),
    subscribe: () => { },
  };
  it('should render a list of Card', () => {
    const wrapper = mount(
      <Router>
        <Provider store={store}>
          <Ads ads={store.getState().ads} history={history} />
        </Provider>
      </Router>
    );
    expect(wrapper.find('Card')).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
});
