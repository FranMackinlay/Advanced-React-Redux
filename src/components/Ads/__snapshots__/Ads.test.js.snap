// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ads should render a list of Card 1`] = `ReactWrapper {}`;

exports[`Ads should render a list of Cards 1`] = `ReactWrapper {}`;
